workers/unity/Assets/Playground/Scripts/Worker/MobileClientWorkerConnector.cs

```diff
@@ -1,20 +1,49 @@
+using System;
+using System.Net;
 using Improbable.Gdk.Core;
 using Improbable.Gdk.Mobile;
+using Improbable.Worker.CInterop;
 using UnityEngine;
 
 namespace Playground
 {
-    public class MobileClientWorkerConnector : DefaultMobileWorkerConnector
+    public class MobileClientWorkerConnector : WorkerConnector, IMobileSettingsProvider
     {
 #pragma warning disable 649
         [SerializeField] private GameObject level;
+        [SerializeField] private string ipAddress;
 #pragma warning restore 649
 
         private GameObject levelInstance;
 
         public async void Start()
         {
-            await Connect(WorkerUtils.MobileClient, new ForwardingDispatcher()).ConfigureAwait(false);
+            var connParams = new ConnectionParameters
+            {
+                WorkerType = WorkerUtils.MobileClient,
+                DefaultComponentVtable = new ComponentVtable()
+            };
+
+            var flowSettingsProvider = new MobileConnectionFlowModifier(new PlayerPrefsSettingsProvider(), new CommandLineSettingsProvider(), this);
+
+            var builder = new SpatialOSConnectionHandlerBuilder()
+                .SetConnectionParameters(connParams)
+                .SetConnectionFlowModifier(flowSettingsProvider)
+                .SetConnectionParametersModifier(new MobileConnectionParametersModifier());
+
+            switch (flowSettingsProvider.GetConnectionService())
```

Comment:
one thing that I'm still wondering: would it be possible to hide that from the user?
just let them set whichever flows they want and then handle the switch-case in the builder? or do we not want to do it?

Reply:
Hmm so I guess doing that would remove the simplest use case which is something like: 

```csharp
var builder = new SpatialOSConnectionHandlerBuilder()
	.SetConnectionParameters(connParams)
	.SetConnectionFlow(new ReceptionistFlow());

await Connect(builder, ...)
```

That is to say - there is no requirement currently that you provide anything beyond the two things above. 

---

Also where does the value from the switch-case come from? What do we do for the mobile case where the standard Locator flow is entirely unsupported? Does a user still provide the flow / just set it to null? 

I guess my main objection is that this PR greatly reduces the amount of magic that happens, at the cost of pushing a bit more logic up to the user code layer. Everything is opt in as opposed to opt out (status quo).

Reply:
quick idea, not completely thought trough: 
if the flow is not set and it will be selected for some reason, it should throw an exception,. e.g. a "default does-not-exist flow" object. you could instead have a 
`builder.SetConnectionServiceHandler(yourObject)` that implements the logic for the switch. that way you still expose it, but it looks more like a basic builder pattern and is easier to follow in user code.

Reply:
Discussed offline and reworked slightly